src/Lean/Compiler/LCNF/JoinPoints.lean

```diff
@@ -5,11 +5,94 @@ Authors: Henrik Böving
 -/
 import Lean.Compiler.LCNF.CompilerM
 import Lean.Compiler.LCNF.PassManager
+import Lean.Compiler.LCNF.PullFunDecls
 
 namespace Lean.Compiler.LCNF
 
+-- TODO: These can be used in a much more general context
```

Comment:
Good point. Let's create a `MapFVar` file at LCNF.
I think we should focus on LCNF only. Thus, we should not care whether the `Expr` has a lot of sharing or not.